src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -52,38 +60,34 @@ internal abstract class AbstractSnippetInfoService : ForegroundThreadAffinitized
                 if (textManager.GetExpansionManager(out _expansionManager) == VSConstants.S_OK)
                 {
                     ComEventSink.Advise<IVsExpansionEvents>(_expansionManager, this);
+                    _waiter = new AggregateAsynchronousOperationListener(asyncListeners, FeatureAttribute.Snippets);
+                    _languageGuidForSnippets = languageGuidForSnippets;
+                    PopulateSnippetCaches();
```

Comment:
It looks like this is getting called without setting or verifying `_expansionManager != null` which you check in `OnAfterSnippetsUpdate()` below.  Are you missing a `null` check or am I missing the initialization of `_expansionManager` somewhere else?